### PR TITLE
Split content of interface.ts

### LINKED ISSUE #4: Split Interfaces.ts contents
#### Describe your issue

The file interfaces.ts currently contains all interfaces, enums, discoveryKeys, types, etc. that are implemented in core_contracts.

#### How can others reproduce the issue?

Look at the interfaces.ts file.

#### Possible solution

Move each interface, enum, etc. into a file of its own.

#### Issue checklist

Please check the boxes in this list after submitting your Issue:

- [x] I've checked if this issue already exists
- [x] I've included all the information that i think is relevant
- [x] I've added logs and/or screenshots (if applicable)
- [x] I've mentioned PRs and issues that relate to this one

### PR DESCRIPTION
**Pull request by ElRaptorus**
Closes #4 

## What did you change?

- All interfaces, enums, asf. are now located in separate files.
- Interfaces and enums have been grouped into UseCase-specific folders (where applicable).

## How can others test the changes?

- Look through the source folder, you'll immediately see the changes.
- Build a solution that uses the core_contracts project. The build should run exactly as before, since all changes made were exclusively cosmetic and non-functional. 

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.)
- [x] I've merged the `develop` branch into my branch before finishing this PR.
- [x] I've **not added any other changes** than the ones described above.
- [x] I've mentioned all **PRs, which relate to this one**

### CODE REVIEW

File: src/entity/ientity.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {ExecutionContext} from '../execution_context';
+import {IPrivateRemoveOptions, IPrivateSaveOptions, IPublicGetOptions} from '../options';
+import {IFromPojoOptions, IToPojoOptions} from '../pojo';
+import {IPrivateQueryOptions} from '../query';
```
Comment (heikomat) [inline src/entity/ientity.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

File: src/execution_context.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import {IIamService, IIdentity, ISessionStoreData, TokenType} from './interfaces';
+import {IIamService, IIdentity, TokenType} from './iam';
```
Comment (heikomat) [inline src/execution_context.ts]: Import from the index-file (`./iam/index`)

File: src/iam/iiam_service.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import {ExecutionContext} from '../execution_context';
+import {IAuthObject} from './iauth_object';
+import {IIdentity} from './iidentity';
+import {IUserEntity} from './iuser_entity';
+import {TokenType} from './token_type';
```
Comment (heikomat) [inline src/iam/iiam_service.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

File: src/inaming_convention_service.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import {IExpand} from './iexpand';
+import {IRegistrationKey} from './iregistration_key';
+import {EntityNamespaceTypeKey, EntityTypeKey, EntityTypeServiceKey, SchemaKey, TypeKey} from './type_aliases';
```
Comment (heikomat) [inline src/inaming_convention_service.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

File: src/options/ipublic_get_options.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import {IExpand} from '../iexpand';
+import {IQueryObject} from '../query';
```
Comment (heikomat) [inline src/options/ipublic_get_options.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

File: src/query/icalculated_property_on_query_callback.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import {ExecutionContext} from '../execution_context';
+import {IQueryClause} from './iquery_clause';
+import {IQueryObject} from './iquery_object';
```
Comment (heikomat) [inline src/query/icalculated_property_on_query_callback.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

File: src/schema/ischema.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import {IExpand} from '../iexpand';
+import {IIndexes} from '../iindexes';
+import {EntityTypeKey} from '../type_aliases';
```
Comment (heikomat) [inline src/schema/ischema.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

File: src/sorting/icalculated_property_on_sort_callback.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import {ISortOptions} from './isort_options';
+import {ISortProperty} from './isort_property';
```
Comment (heikomat) [inline src/sorting/icalculated_property_on_sort_callback.ts]: please import from the barrel-index-file of a folder if you import multiple interfaces from the same folder of this module

**Comment by ElRaptorus** [comment]
@heikomat done

**Pull request merged**